### PR TITLE
87403 delete DLQ reader during subscriber removal

### PR DESCRIPTION
**Pull request by dawidkeshop**

### CODE REVIEW

File: src/CaptainHook.Application/Handlers/Subscribers/DeleteSubscriberRequestHandler.cs [hunk 1]
```diff
@@ -53,10 +59,31 @@ private async Task<OperationResult<SubscriberDto>> DeleteSubscriberAsync(DeleteS
...
+                return existingItem.Error;
+            }
+
+            var subscriberEntity = existingItem.Data;
+
+            return await _entityToConfigurationMapper.MapToWebhookAsync(subscriberEntity)
+                .Then(async webhookConfig => await _directorService.CallDirectorService(new DeleteReader(webhookConfig)))
+                .Then<SubscriberConfiguration, SubscriberEntity>(async _ => await DeleteDlqIfExists(subscriberEntity))
+                .Then(async _ => await _subscriberRepository.RemoveSubscriberAsync(subscriberId))
+                .Then<SubscriberId, SubscriberDto>(_ => _entityToDtoMapper.MapSubscriber(subscriberEntity));
```
Comment (tabman83) [inline src/CaptainHook.Application/Handlers/Subscribers/DeleteSubscriberRequestHandler.cs:74]: Super! Don't we have an overload of Then which accepts task-async ?

File: src/CaptainHook.Application/Handlers/Subscribers/DeleteSubscriberRequestHandler.cs [hunk 2]
```diff
@@ -53,10 +59,31 @@ private async Task<OperationResult<SubscriberDto>> DeleteSubscriberAsync(DeleteS
...
+        }
+
+        private async Task<SubscriberEntity> DeleteDlqIfExists(SubscriberEntity subscriberEntity)
+        {
+            if (subscriberEntity.HasDlqHooks)
+            {
+                return await _entityToConfigurationMapper.MapToDlqAsync(subscriberEntity)
+                    .Then(async dlqConfig => await _directorService.CallDirectorService(new DeleteReader(dlqConfig)))
```
Comment (tabman83) [inline src/CaptainHook.Application/Handlers/Subscribers/DeleteSubscriberRequestHandler.cs:82]: Same as before

**Review by tabman83 — APPROVED** [review]
See the comments above

File: src/CaptainHook.Application/Handlers/Subscribers/DeleteSubscriberRequestHandler.cs (hunk 1 above)
Comment (dawidkeshop) [reply src/CaptainHook.Application/Handlers/Subscribers/DeleteSubscriberRequestHandler.cs:74]: Sure, you're right!

**Pull request merged**